Fix Aggregation Pushdown docs (#6277)
prestosql/presto: presto-docs/src/main/sphinx/optimizer/pushdown.rst

Support for pushdown is specific to each connector and the relevant underlying
database or storage system.

Aggregation Pushdown
--------------------

Aggregation pushdown depends on a number of factors:

> * general support for aggregation pushdown in the connector
> * function-specific support for pushdown in the connector
> * query that allows the pushdown to take place

Reviewer: ```suggestion
* If aggregation pushdown is generally supported by the connector.
* If pushdown of specific function is supported by the connector.
* If query structure allows pushdown to take place.
```

Reviewer 2: I would not do that unless you also change the sentence in front of that list .. with this change it would be wrong

Reviewer: It reads fine to me - but I assume you are right :)
As for enumeration, the google documentation style guide recommends making each of the points a capitalized sentence ending with a dot.

Reviewer 2: Mostly true .. so yes, maybe update the opening sentence as well and then go with this

Reviewer: Like I said above, I do not see why the current one is wrong (it does not mean it is ok). Can you propose sth which makes this correct English paragraph :)

Reviewer 2: Aggregation push down relies on the following conditions to be true:

or 

Aggregation push down relies on the following conditions to be fulfilled:

Reviewer 2: And you would have to do

* If aggregation pushdown is generally supported by the connector.
* If pushdown of the specific function is supported by the connector.
* If the query structure allows pushdown to take place.

Reviewer: Thank you :)

> The best way to analyze if aggregation pushdown for a specific query is performed is to
> take a closer look at the :doc:`EXPLAIN plan </sql/explain>` of the query. If an aggregate
> function is successfully pushed down to the connector, the explain plan does **not** show
> that ``Aggregate`` operator. The explain plan only shows the operations that are performed
> by Presto.

Reviewer: ```suggestion
You can check if pushdown for a specific query is performed by looking at the :doc:`EXPLAIN plan </sql/explain>` of the query. 
If an aggregate function is successfully pushed down to the connector, the explain plan does **not** show
that ``Aggregate`` operator. The explain plan only shows the operations that are performed by Presto.
```

Reviewer 2: Breaking up the lines and reformatting like this does NOT affect output and violates our line length standard (80 char). Please keep it as is

Reviewer: Sure, not forcing anything. Not my area of expertise.
